plutarch-trace/README.md

```diff
@@ -0,0 +1,149 @@
+# plutarch-trace
+
+## Usage
+You **need** `cabal-install >= 3.0` for this. Your package's cabal file **must** use a `cabal-version` of 3.0 or higher.
+
+* Add `plutarch-trace` to your `build-depends`.
+
+The next steps depend on whether or not you want tracing enabled.
+### Enable Tracing
+* Add `plutarch-trace:enable` to your `build-depends`.
+* Add a mixin similar to the following-
+
+  ```cabal
+  plutarch-trace (Plutarch.Trace) requires (Plutarch.TraceSig as Plutarch.Trace.Enable),
+  ```
+
+  This makes the `Plutarch.Trace` module available for you to access in your code - and it sets it up with the real tracing functions.
+
+  You can rename the `Plutarch.Trace` module if you want, to something else-
+  ```cabal
+  plutarch-trace (Plutarch.Trace as FooBar) requires (Plutarch.TraceSig as Plutarch.Trace.Enable),
+  ```
+  This makes it so you import `Plutarch.Trace` as `FooBar`. This *will* hide the `Plutarch.Trace` module, meaning you can only import it as `FooBar`. Otherwise it's the exact same as just importing `Plutarch.Trace` without renaming.
+
+Example `.cabal` file-
+```cabal
+cabal-version: 3.0
+name: foo
+version: 1.0.0
+
+executable foo-exe
+  main-is:
+    Main.hs
+  build-depends:
+    base,
+    plutarch-trace,
+    plutarch-trace:enable
+  mixins:
+    plutarch-trace (Plutarch.Trace) requires (Plutarch.TraceSig as Plutarch.Trace.Enable)
+  default-language: Haskell2010
+```
+
+### Disable Tracing
+* Add `plutarch-trace:disable` to your `build-depends`.
```

Comment:
You might want to write a few tips about how to make the choice available without changing the source code; I'm specifically thinking that end-users can add a flag to their `.cabal`, then have their executable `if` that flag to choose between enabling and disabling tracing. There's also the option of duplicating the `executable` section such that one has `foo-exe` and `foo-exe-debug`.

Reply:
Done! You may want to review [7592706](https://github.com/Plutonomicon/plutarch/pull/39/commits/7592706c219c7115a9f25b2e89f58acda15f4b49) separately. I also added some details on hiding backpack impl details in case anyone needs that. The cabal user guide on backpack was surprisingly useful for this.

Reply:
LGTM.